scripts/converter_hf_to_mcore.py

```diff
@@ -54,6 +54,37 @@ def __init__(self):
         self.model = ModelConfig()
 
 
+def test_conversion(megatron_model_provider, tfconfig, output_path, model):
+    ########### test ###########
+    # load model
+    model_test = get_model(
+        model_provider_func=megatron_model_provider,
+        model_type=ModelType.encoder_or_decoder,
+        wrap_with_ddp=True,
+        transformer_config=tfconfig,
+    )
+    ref_state_dict = model_test[0].module.sharded_state_dict()
+    dist_checkpointing.load(ref_state_dict, output_path, strict=StrictHandling.ASSUME_OK_UNEXPECTED)
+
+    dut_state_dict = model[0].module.state_dict()
+    for name in dut_state_dict.keys():
+        if dut_state_dict[name] is None:
+            continue
+        dut_data = dut_state_dict[name].data
+        if name in ref_state_dict:
+            ref_data = ref_state_dict[name].data
+            assert dut_data.shape == ref_state_dict.shape, f"{name=} {dut_data.shape=} {ref_data.shape=}"
+            assert (dut_data == ref_data).all(), f"{name} is not equal"
+    for name in ref_state_dict.keys():
+        if ref_state_dict[name] is None:
+            continue
+        ref_data = ref_state_dict[name].data
+        if name in dut_state_dict:
+            dut_data = dut_state_dict[name].data
+            assert dut_data.shape == ref_data.shape, f"{name=} {dut_data.shape=} {ref_data.shape=}"
+            assert (dut_data == ref_data).all(), f"{name} is not equal"
+
```

Comment:
can we add one log to indicate that the conversion testing is passed or similar statement?